src/core/proj/qgscoordinatereferencesystem.cpp

```diff
@@ -3113,6 +3137,36 @@ QgsCoordinateReferenceSystem QgsCoordinateReferenceSystem::toGeographicCrs() con
   }
 }
 
+QgsCoordinateReferenceSystem QgsCoordinateReferenceSystem::toGeocentricCrs() const
+{
+  if ( type() == Qgis::CrsType::Geocentric )
+  {
+    return *this;
+  }
+
+  if ( PJ *obj = d->threadLocalProjObject() )
+  {
+    PJ_CONTEXT *pjContext = QgsProjContext::get();
+
+    QgsProjUtils::proj_pj_unique_ptr datum( proj_crs_get_datum( pjContext, obj ) );
```

Comment:
I think the type of the this CRS should be tested a bit more.
- proj_crs_get_datum() for example will work if this CRS is a vertical CRS, but making a geocentric CRS from that is non-sense.
- if this CRS is a compound CRS, proj_crs_get_datum/get_datum_ensemble will fail.  The horizontal CRS should be extracted first
- if this CRS is a bound CRS, they will also fail. QgsProjUtils::unboundCrs() should be called first

Reply:
Thanks, fixed!